### PR TITLE
+ comp flags, depr init, use new, cleanup

### PR DESCRIPTION
**Pull request by shayanhabibi**

### CODE REVIEW

File: loony/spec.nim [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import std/atomics
 
+const
+  LoonyNodeAlignment {.intdefine.} = 11
+  LoonySlotCount {.intdefine.} = 1024
```
Comment (disruptek) [inline loony/spec.nim]: These need to start with a lowercase letter, right, because convention is to match the package name...

File: loony.nim [hunk 2]
```diff
@@ -208,6 +208,7 @@ proc advHead(queue: LoonyQueue; curr, h, t: var TagPtr): AdvHead =
 
 proc pushImpl[T](queue: LoonyQueue[T], el: T,
                     forcedCoherance: static bool = false) =
+  assert not queue.isNil(), "The queue has not been initialised"
```
Comment (disruptek) [inline loony.nim]: We need to read the reference anyway, so we're just paying the cost of a conditional.  I think we should make this check even in `release` or `danger` builds.

File: README.md [hunk 3]
```diff
@@ -69,7 +71,7 @@ import pkg/loony
 
 type AnyRefObject = ref object
 
-var loonyQueue = initLoonyQueue[AnyRefObject]()
+var loonyQueue = newLoonyQueue[AnyRefObject]()
```
Comment (disruptek) [inline README.md]: Wait, what?  Why does it have to be an `object`?  Or at least, this is how the code reads...

**Review by disruptek — CHANGES_REQUESTED** [review]
All the public symbols, notably `newLoonyQueue`, `push`, `pop`, `unsafePush`, `unsafePop` need documentation.

File: loony.nim (hunk 2 above)
Comment (shayanhabibi) [reply loony.nim]: Fair; I wasn't sure what the cost would be. I guess it's also significantly less than all the atomic writes lol

File: README.md (hunk 3 above)
Comment (shayanhabibi) [reply README.md]: Could you clarify this?

File: loony/spec.nim (hunk 1 above)
Comment (shayanhabibi) [reply loony/spec.nim]: Sure thing; I was just going off what I saw on the Manual for  this pragma

File: README.md (hunk 3 above)
Comment (disruptek) [reply README.md]: Loony can work with any `ref` or `ptr` but for the purposes of demonstration, this type suggests that it's limited to `ref object` just due the choice of name...

**Comment by shayanhabibi** [comment]
Ehhh I'm merging this since the blog has the `newLoonyQueue` api and it'll be odd if it's not actually in yet

**Pull request merged**